File: Baekjoon/Successful/python/1477.py
import math
import copy

n, m, l = map(int, input().split())
if n > 0 : loc = list(map(int, input().split()))
else : loc = []
loc.append(l)
loc.sort()
d = []
d.append(loc[0])
for i in range(1, len(loc)) :
    d.append(loc[i] - loc[i - 1])

c = [0] * len(d)
cd = copy.deepcopy(d)
while m > 0 :
    cds = sorted(cd, reverse=True)
    index = cd.index(cds[0])
    c[index] += 1
    cd[index] = math.ceil(d[index] / (c[index] + 1))
    m -= 1
print(max(cd))

File: Baekjoon/Successful/swift/1477.swift
import Foundation

let input = (readLine() ?? "0 0 100").components(separatedBy: " ")
let n = Int(input[0])!, l = Int(input[2])!
var m = Int(input[1])!
var loc: [Int] = [l], d: [Int] = [Int](), c: [Int] = Array(repeating: 0, count: n + 1)

(readLine() ?? "0").components(separatedBy: " ").forEach{ loc.append(Int($0)!) }
loc.sort()

for (index, dest) in loc.enumerated() {
    guard index > 0 else { 
        d.append(dest) 
        continue
    }
    d.append(dest - loc[index - 1])
}

var cd = d.map{ $0 }
for _ in (0..<m) {
    let index = cd.firstIndex(of: cd.max()!)!
    c[index] += 1
    cd[index] = Int(ceil(Double(d[index]) / Double(c[index] + 1)))
}
print(cd.max()!)